test/setupAfterEnv.ts

```diff
@@ -0,0 +1,74 @@
+const UNRESOLVED = Symbol('timedOut');
+// Store this in case it gets stubbed later
+const originalSetTimeout = global.setTimeout;
+const TIME_TO_WAIT_UNTIL_UNRESOLVED = 100;
+
+/**
+ * Produces a sort of dummy promise which can be used in conjunction with a
+ * "real" promise to determine whether the "real" promise was ever resolved. If
+ * the promise that is produced by this function resolves first, then the other
+ * one must be unresolved.
+ *
+ * @param duration - How long to wait before resolving the promise returned by
+ * this function.
+ * @returns A promise that resolves to a symbol.
+ */
+const treatUnresolvedAfter = (duration: number): Promise<typeof UNRESOLVED> => {
+  return new Promise((resolve) => {
+    originalSetTimeout(resolve, duration, UNRESOLVED);
+  });
+};
+
+expect.extend({
+  /**
+   * Tests that the given promise is never fulfilled or rejected past a certain
+   * amount of time (which is the default time that Jest tests wait before
+   * timing out as configured in the Jest configuration file).
+   *
+   * Inspired by <https://stackoverflow.com/a/68409467/260771>.
+   *
+   * @param promise - The promise to test.
+   * @returns The result of the matcher.
+   */
+  async toNeverResolve(promise: Promise<any>) {
+    if (this.isNot) {
+      throw new Error(
+        'Using `.not.toNeverResolve(...)` is not supported. ' +
+          'You probably want to either `await` the promise and test its ' +
+          'resolution value or use `.rejects` to test its rejection value instead.',
+      );
+    }
+
+    let resolutionValue: any;
+    let rejectionValue: any;
+    try {
+      resolutionValue = await Promise.race([
+        promise,
+        treatUnresolvedAfter(TIME_TO_WAIT_UNTIL_UNRESOLVED),
+      ]);
+    } catch (e) {
+      rejectionValue = e;
+    }
+
+    return resolutionValue === UNRESOLVED
+      ? {
+          message: () =>
+            `Expected promise to resolve after ${TIME_TO_WAIT_UNTIL_UNRESOLVED}ms, but it did not`,
+          pass: true,
+        }
+      : {
+          message: () => {
+            return `Expected promise to never resolve after ${TIME_TO_WAIT_UNTIL_UNRESOLVED}ms, but it ${
+              rejectionValue
+                ? `was rejected with ${rejectionValue}`
+                : `resolved with ${resolutionValue}`
+            }`;
+          },
+          pass: false,
+        };
+  },
+});
+
+// Export something so that TypeScript thinks that we are performing type
+// augmentation
+export {};
```

Comment:
This line was necessary for type augmentation, but now that that's been extracted to another file, this line is no longer necessary in this file.

Reply:
Hmm. Still seems to be required by `import/unambiguous`

Reply:
Oh, okay. Then the comment is wrong, because we're not performing type augmentation here. Would it would be more appropriate in this case to put `/* eslint-disable import/unambiguous */` at the top of the file?

Reply:
Good point. At the very least we should make the comment more accurate.

I was thinking that ensuring our modules were identifiable as "Unambiguous JavaScript Grammer" was desirable in general, to ensure we don't get surprised when some tool gets confused. I don't know for certain that it would be relevant for us, but it seems reasonably easy to accommodate.

Reply:
Comment fixed here: https://github.com/MetaMask/eth-json-rpc-middleware/pull/191/commits/66c03e43f328dad1e549b59639ffae7796bcffa8